### ISSUE TITLE
How to access openseadragon instance in other files

### ISSUE THREAD

**Issue by johnguirgis** [openseadragon/openseadragon#1983]
I am trying to implement Annotorious with openseadragon but I need an instance of the openseadragon viewer in order to setup the annotations. My issue is the viewer is initialized in another file, so is there a way I can get a reference to it so that I can initialize Annotorious? I am new to web development so this may be a simple task.

**Comment by iangilman** [comment]
Is the existing viewer already stored in a variable somewhere? Do you have access to that part of the code? If possible, that's definitely how you want to do it. For instance, if in the file where the viewer is created, you say `window.osdViewer = viewer` or whatever, then it'll be global and you can access it from your file. That's just one example... there are lots of ways to get variables from one file to another. 

Otherwise, here's a hack you can try: https://stackoverflow.com/questions/67396654/get-reference-to-existing-openseadragon-viewer

**Comment by johnguirgis** [comment]
Thanks for your response. Unfortunately I don't have access to the viewer variable. I tried the hack you suggested but I am getting that the viewer is undefined.

**Comment by iangilman** [comment]
@johnguirgis whoops! Looks like I forgot to add `prototype` there... I've updated the code on StackOverflow; please try that and see if it works!

**Comment by iangilman** [comment]
I'm marking this issue as `enhancement`, since it would be nice to have an explicit way to do this without having to hack it. I'm thinking something like:

```
var viewer = OpenSeadragon.getViewer(element);
```

Anyone who wants to tackle this, I'm happy to point you in the right direction. Probably wouldn't be a terribly challenging fix to make.

**Comment by johnguirgis** [comment]
Thanks for your followup. Again, `viewer` becomes undefined after the `isOpen()` call :confused:

**Comment by iangilman** [comment]
Strange that it's not working for you! Have you tried putting a console.log in the function to see if it gets called? I wonder if your `OpenSeadragon` is available to your code file? Is it loaded as a global, or is it imported into the other file? 

At any rate, I made a test CodePen: https://codepen.io/iangilman/pen/eYvVrgm?editors=1111 ... and that seems to be working. There must be something about how your code is structured that's creating the disconnect.

**Comment by johnguirgis** [comment]
It is indeed being called and it is initially working, but it is behaving usually when I try to add or modify annotations.

**Comment by HerCerM** [comment]
Hi @iangilman, I would like to take this issue

I'm thinking of implementing the functionality according to the API you propose:

```
var viewer = OpenSeadragon.getViewer(element);
```

It seems that the function should be implemented in [this object in openseadragon.js](https://github.com/openseadragon/openseadragon/blob/fd85d5e22f8a8cf9f0b534c2428f4c7f645268bf/src/openseadragon.js#L1116)

**Comment by HerCerM** [comment]
@iangilman I opened a PR with my working implementation. I'm all ears for any thoughts

**Comment by iangilman** [comment]
@HerCerM excellent! Thank you for tackling this :)

**Comment by iangilman** [comment]
Fixed by #2000
